### DISCUSSION TITLE
Setting Crystal Axes Instead of Planes for Pole Figures

### DISCUSSION THREAD

**Discussion by beroshcr** [mtex-toolbox/mtex#1457] [Ask Anything] [answered]
Hi!

I am trying to figure out how to plot a pole figure for the c-axis of a monoclinic augite crystal. I want to set the c-axis of the crystal to equal the (001) direction. 

Any ideas how to do this in MTEX?

Thanks!

- Christian

**Comment by lfgmorales** [comment, marked as answer]
Try the option ‘uvw’ after you define Miller(0,0,1)

Best

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 17 Jun 2022, at 03:23, beroshcr ***@***.***> wrote:

﻿

Hi!

I am trying to figure out how to plot a pole figure for the axis of a monoclinic augite crystal. I want to set the axis of the crystal to equal the (001) direction.

Any ideas how to do this in MTEX?

Thanks!

  *   Christian

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1457>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN3EOKLPTSZTCNVHB73VPPHQ7ANCNFSM5ZAUHT6A>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1457", "url": "https://github.com/mtex-toolbox/mtex/discussions/1457", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by beroshcr** [reply]
Thanks Luiz it worked! 

However I do have another question if you don't mind. Right now I am plotting indexed pixels of augite onto my pole figure. I was wondering if there was a specific way to plot the orientations of my actual grain structures of augite (not pixels) onto the pole figure. 

I understand I should create a "grains" variable. For example: 

% The reconstructed grains are stored in the variable "grains". 
grains = calcGrains(ebsd,'angle',10*degree)

But I don't seem to understand how to plot them on a pole figure. I keep receiving errors.

Thanks again. 

Christian

**Comment by lfgmorales** [comment]
Hi Christian,

You have to calculate an odf out of your grains(‘augite’).meanOrientation then plot the pole figures as you would plot the ones for the whole ebsd dataset. Important here is your grain definitions

Let me know if that works

Cheers

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 17 Jun 2022, at 19:32, beroshcr ***@***.***> wrote:

﻿

Thanks Luiz it worked!

However I do have another question if you don't mind. Right now I am plotting indexed pixels of augite onto my pole figure. I was wondering if there was a specific way to plot the orientations of my actual grain structures of augite (not pixels) onto the pole figure.

I understand I should create a "grains" variable. For example:

% The reconstructed grains are stored in the variable "grains".
grains = calcGrains(ebsd,'angle',10*degree)

But I don't seem to understand how to plot them on a pole figure. I keep receiving errors.

Thanks again.

Christian

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1457#discussioncomment-2973402>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN3CYX5WHKYS7MAFMJ3VPSZBTANCNFSM5ZAUHT6A>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1457#discussioncomment-2973402", "url": "https://github.com/mtex-toolbox/mtex/discussions/1457#discussioncomment-2973402", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by beroshcr** [reply]
Hi Luiz,

It worked! Thank you. However, I have one final question in terms of merging twinned boundaries between two grains. My script seems to remove the boundaries fine enough but my GROD values are suspicious. I am wondering where I've gone wrong here, they are drastically higher after completing the twinning analysis. I suspect I am inputting something wrong. 

Additionally, I set the threshold for the grain boundaries between 1-10 degrees but the GROD values eclipse these values drastically. I don't understand how we can see values >10 if I've set the max degree of 10 to indicate a new grain. 

%% 
[grains, ebsd.grainId] = calcGrains(ebsd_merged('indexed'),'angle',1*degree,10*degree);
ebsd(grains(grains.grainSize<5)) = [];
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',1*degree,10*degree);

*Start* 

%% Twinning Analysis and Merging

CS = ori.CS

gB = grains.boundary

gB_Au = gB('Augite','Augite')

plot(gB_Au,gB_Au.misorientation.angle./degree,'linewidth',2)
mtexColorbar

close all
histogram(gB_Au.misorientation.angle./degree,40)
xlabel('misorientation angle (degree)')
%% 

ind = gB_Au.misorientation.angle>175.7*degree & gB_Au.misorientation.angle<180*degree;
mori = gB_Au.misorientation(ind);

scatter(mori)

% determine the mean of the cluster
mori_mean = mean(mori,'robust')

% determine the closest special orientation relation ship
round2Miller(mori_mean)

twinning = orientation.map(Miller(-2,0,-3,CS),Miller(-3,0,2,CS),...
  Miller(0,-1,0,CS),Miller(0,1,0,CS))

% the rotational axis
round(twinning.axis)

% the rotational angle
twinning.angle / degree

isTwinning = angle(gB_MgMg.misorientation,twinning) < 10*degree;
twinBoundary = gB_MgMg(isTwinning)

plot(grains)
%plot(ebsd('indexed'),ebsd('indexed').orientations)
hold on
%plot(gB_MgMg,angle(gB_MgMg.misorientation,twinning),'linewidth',4)
plot(twinBoundary,'linecolor','w','linewidth',0.1,'displayName','twin boundary')
hold off

[mergedGrains,parentId] = merge(grains,twinBoundary);

hold on
plot(mergedGrains.boundary,'linecolor','k','linewidth',2.5,'linestyle','-',...
  'displayName','merged grains')
hold off

plot(mergedGrains,'linecolor','k','linewidth',2.5,'linestyle','-',...
  'displayName','merged grains')
hold off

plot(mergedGrains,mergedGrains.GOS ./ degree)

ori = mergedGrains('Augite').meanOrientation
h = Miller({1,0,0,'uvw'},{0,1,0,'uvw'},{0,0,1,'uvw'},ori.CS);
plotPDF(ori,h,'contourf', 'all')
mtexColorbar
%% 

childs = grains(parentId == mergedGrains(534,535,536).id);

% this averages the GOS of the child grains into the parent grains
mergedGrains.prop.GOS = accumarray(parentId,grains.GOS,size(mergedGrains),@nanmean);

% extract GOS and area
childGOS = grains.GOS;
childArea = grains.area;

% compute the weighted averages
mergedGrains.prop.GOS = accumarray(parentId,1:length(grains),size(mergedGrains),...
  @(id) nanmeanWeights(childGOS(id),childArea(id)));

plot(mergedGrains,mergedGrains.GOS  ./ degree)
mtexColorbar

ebsd_merged = ebsd;

% update the grainIds to the parentIds
ebsd_merged('Augite').grainId = parentId(grains.id2ind(ebsd('Augite').grainId))
%% 

grod = ebsd_merged.calcGROD(mergedGrains);
plot(ebsd_merged,grod.angle./degree,'micronbar','on')
mtexColorbar('title','misorientation angle to meanorientation in degree')
mtexColorMap LaboTeX

*End*
![mergedGROD](https://user-images.githubusercontent.com/57849453/174494037-cbda1ae5-78e8-4abf-b41b-b7925b4980f4.jpg)
![unmergedGROD](https://user-images.githubusercontent.com/57849453/174494112-08eed04a-8853-4745-827b-e18114247a05.jpg)
